compiler/fir/resolve/src/org/jetbrains/kotlin/fir/resolve/providers/impl/FirTypeResolverImpl.kt

```diff
@@ -90,14 +96,38 @@ class FirTypeResolverImpl(private val session: FirSession) : FirTypeResolver() {
         }
     }
 
+    private fun FirQualifierResolver.resolveEnumEntrySymbol(
+        qualifier: List<FirQualifierPart>,
+        classId: ClassId
+    ): FirVariableSymbol<FirEnumEntry>? {
+        // Assuming the current qualifier refers to an enum entry, we drop the last part so we get a reference to the enum class.
+        val enumClassSymbol = resolveSymbolWithPrefix(qualifier.dropLast(1), classId) ?: return null
+        val enumClassFir = enumClassSymbol.fir as? FirRegularClass ?: return null
+        if (!enumClassFir.isEnumClass) return null
+        val enumEntryMatchingLastQualifier = enumClassFir.declarations
+            .filterIsInstance<FirEnumEntry>()
+            .firstOrNull { it.name == qualifier.last().name }
```

Comment:
The following code probably more optimal:

```suggestion
            .firstOrNull { it is FirEnumEntry && it.name == qualifier.last().name }
```

Reply:
Sounds good. Thanks!